src/popover-helpers.ts

```diff
@@ -354,10 +454,62 @@ function closeAllOpenPopovers(
   focusPreviousElement = false,
   fireEvents = false,
 ) {
-  let popover = topMostAutoPopover(document);
+  let popover = topMostAutoOrHintPopover(document);
   while (popover) {
     hidePopover(popover, focusPreviousElement, fireEvents);
-    popover = topMostAutoPopover(document);
+    popover = topMostAutoOrHintPopover(document);
+  }
+}
+
+function closeAllOpenPopoversInList(
+  list: Set<HTMLElement>,
+  focusPreviousElement = false,
+  fireEvents = false,
+) {
+  let popover = topMostPopoverInList(list);
+  while (popover) {
+    hidePopover(popover, focusPreviousElement, fireEvents);
+    popover = topMostPopoverInList(list);
+  }
+}
+
+// https://html.spec.whatwg.org/#hide-popover-stack-until
+function hidePopoverStackUntil(
+  endpoint: Element | Document,
+  set: Set<HTMLElement>,
+  focusPreviousElement: boolean,
+  fireEvents: boolean,
+) {
+  let repeatingHide = false;
+  let hasRunOnce = false;
+  // TODO: This doesn't match the spec changes in
+  // https://github.com/whatwg/html/pull/9778/files#diff-41cf6794ba4200b839c53531555f0f3998df4cbb01a4d5cb0b94e3ca5e23947dR85830
+  while (repeatingHide || !hasRunOnce) {
+    hasRunOnce = true;
+    let lastToHide = null;
+    let foundEndpoint = false;
+    for (const popover of set) {
+      if (popover === endpoint) {
+        foundEndpoint = true;
+      } else if (foundEndpoint) {
+        lastToHide = popover;
+        break;
+      }
+    }
+    if (!lastToHide) return;
+    while (getPopoverVisibilityState(lastToHide) === 'showing' && set.size) {
+      hidePopover(
+        [...set].pop() as HTMLElement,
```

Comment:
I'm using this technique in a couple of places to get the last item in a Set -- maybe we should abstract it out into a named function?